src/components/application_manager/test/policy_handler_test.cc

```diff
@@ -311,14 +323,75 @@ TEST_F(PolicyHandlerTest, OnPermissionsUpdated_method_With2Parameters) {
   // Check expectations
   EXPECT_CALL(app_manager_, application_by_policy_id(_))
       .WillOnce(Return(mock_app_));
-  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId_));
-  EXPECT_CALL(*MockMessageHelper::message_helper_mock(),
-              SendOnPermissionsChangeNotification(kAppId_, _, _));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
   // Act
   Permissions perms;
   policy_handler_.OnPermissionsUpdated(kPolicyAppId_, perms);
 }
 
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_TwoParams_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(_, _, _)).Times(0);
+
+  Permissions permissions;
+  policy_handler_.OnPermissionsUpdated(kPolicyAppId_, permissions);
+}
+
+TEST_F(PolicyHandlerTest, OnPermissionsUpdated_InvalidApp_UNSUCCESS) {
+  utils::SharedPtr<application_manager_test::MockApplication> invalid_app;
+  EXPECT_CALL(app_manager_, application_by_policy_id(_))
+      .WillOnce(Return(mock_app_))
+      .WillOnce(Return(invalid_app));
+  EXPECT_CALL(*mock_app_, app_id()).WillOnce(Return(kAppId1_));
+  EXPECT_CALL(mock_message_helper_,
+              SendOnPermissionsChangeNotification(kAppId1_, _, _));
+
```

Comment:
@okozlovlux I see that you have two unsuccessed tests and first you expect call SendOnPermissionsChangeNotification times(0) here you don't use times(0). Is it correct?